### ISSUE TITLE
How to download 1080p from youtube?

### ISSUE THREAD

**Issue by Cinerar** [e-alfred/ocdownloader#29]
No matter what i use Aria or CURL, only 720p is downloaded.
Is there an option to downloade 1080p? 
As long as i understand youtube-dl is used ONLY to get direct link, not for downloading.

**Comment by Cinerar** [comment]
when using youtube-dl  
youtube-dl +f bestvideo+bestaudio url  can be used to download best avialible formats and merge them in one file after they are downloaded. 
Shouldn't ocdownloader rely on youtube-dl for downloading from youtube and other sites, not on aria2c or curl, this will open support for huge range of sites https://rg3.github.io/youtube-dl/supportedsites.html

**Comment by e-alfred** [comment]
Yes, using youtube-dl directly would be much better, the original author changed this behaviour long ago, but it could be patched back in.

The problem with downloads from Youtube with a higher resolution is the fact that only videos up to 720p are provided as a single file. Higher resolutions are served with separate audio and video files that are combined after downloading using ffmpeg by youtube-dl.

**Comment by Cinerar** [comment]
Yes i know that need i believe the ability to download from a lot of other sites could be positive enough to patch this back.  Thanks for the info about changing this behavior will check it out

**Comment by e-alfred** [comment]
The commit can be found here: https://github.com/e-alfred/ocdownloader/commit/f0ce1fb34fad8be0346073dff3f0e6ef20e25475

**Comment by hifihedgehog** [comment]
Any update on the progress of this and ticket #89?

**Comment by e-alfred** [comment]
I started some work in a separate branch but it is far from finished:

https://github.com/e-alfred/ocdownloader/tree/youtubedl-rework

**Comment by NormandoHall** [comment]
> I started some work in a separate branch but it is far from finished:
> 
> https://github.com/e-alfred/ocdownloader/tree/youtubedl-rework

Very interested on this enhancement. Thanks!

**Comment by tsalwierz** [comment]
What's the progress on this? I can use youtube-dl directly, but it's easier to use this app.

**Comment by jsellin78** [comment]
is mp3 format supported?

**Comment by boistordu** [comment]
What's the status on that ? Because aria is still being used apparently because and this is directly impacting what we can actually achieve with downloading video or audio from websites as YouTube or others as shown in
